python/tank/util/user_settings.py

```diff
@@ -45,48 +45,83 @@ def _init_singleton(self):
         logger.debug("Default site: %s" % (self.default_site or "<missing>",))
         logger.debug("Default login: %s" % (self.default_login or "<missing>",))
 
-        self._settings_proxy = self._get_settings_proxy()
-        self._system_proxy = None
-        if self._settings_proxy:
-            logger.debug("Shotgun proxy (from settings): %s" % self._get_filtered_proxy(self._settings_proxy))
-        else:
-            self._system_proxy = self._get_system_proxy()
+        # Retrieve the operating system http proxy string,
+        # or None when it is not defined.
+        self._system_proxy = self._get_system_proxy()
+
+        # Retrieve the settings http proxy string (possibly an empty string),
+        # or None when it is not present in the file.
+        self._settings_proxy = self._get_value("http_proxy")
+
+        if self._settings_proxy is None:
+            # Since the http proxy is not present in the settings file,
+            # fall back on the operating system http proxy.
             if self._system_proxy:
                 logger.debug("Shotgun proxy (from system): %s" % self._get_filtered_proxy(self._system_proxy))
             else:
                 logger.debug("Shotgun proxy: <missing>")
+        else:
+            if self._settings_proxy:
```

Comment:
This is really hard to read for me even when i know the logic:

```
if self._settings_proxy is None:
   ...
else:
   if self._settings_proxy:
      ...
   else:
      ...
```

Can we just remove the `<missing>` and `<empty>` messages? I find them quite confusing. And instead report back from the user: "This is the proxy server shotgun will use when it connects. This is the proxy server that toolkit will use when it connects to the app store. " that sort of thing?

I also find it mildly confusing with `self._system_proxy` and `self._settings_proxy` and `self._app_store_proxy`. Could we name them in a way which made it more clear? the app store proxy comes from the settings too, right? and the system proxy comes from the system settings, which is also a kind of setting. Maybe the word setting could be avoided?